src/test/rustdoc-json/reexport_method_from_private_module.rs

```diff
@@ -0,0 +1,24 @@
+// Regression test for <https://github.com/rust-lang/rust/issues/102583>.
+
+// @has "$.index[*][?(@.name=='missing_from_rustdoc_json')]"
```

Comment:
Nit: Could you also check that S has an impl block, and the impl block has just the expected method

Reply:
So, for whatever reason I can't do:

```
// @has "$.index[*][?(@.kind=='impl' && @.inner.for.inner.name=='S')]"
```

but this one works:

```
// @has "$.index[*][?(@.kind=='impl')].inner.for.inner.name" '"S"'
```

I wanted to use it to use to first count the number of impl blocks for `S` and then count the number of items in this impl. But since it's not possible, I'll just leave it as is I guess...

Reply:
```
// @has "$.index[*][?(@.kind=='impl' && @.inner.for.inner.name=='S')]"
```

doesn't work due to https://github.com/freestrings/jsonpath/issues/91

The way I to work around this is to add a doc comment to the impl and select on that, eg

```rust
pub struct S;

//// impl S
impl S {
    pub fn is_present() {}
}

// @set impl_S = "$.index[*][?(@.docs=='impl S')].id"
// @set is_present = "$.index[*][?(@.name=='is_present')].id"
// @is "$.index[*][?(@.name=='S')].inner.impls[*]" $impl_S
// @is "$.index[*][?(@.docs=='impl S')].inner.items[*]" $is_present
```

Reply:
Smart trick! That doesn't allow to see how many impl blocks are implemented on `S` unfortunately but it's much better than currently. :)

Reply:
If theirs a second impls on `S`, then it'll fail because several things match to `// @is "$.index[*][?(@.name=='S')].inner.impls[*]" $impl_S`, but because this in `#[no_core]`, their are no auto traits.

Similarly, this checks theirs only 1 method in the impl.

Reply:
Makes sense, thanks for the explanation! Then I guess it's ready for final review. :)